src/core/cuda_memory_manager.h

```diff
@@ -76,8 +76,13 @@ class CudaMemoryManager {
   static void Reset();
 
  private:
-  CudaMemoryManager(bool has_allocation) : has_allocation_(has_allocation) {}
+  CudaMemoryManager(bool has_allocation)
+      : has_allocation_(has_allocation), initialized_(false)
+  {
+  }
+
   bool has_allocation_;
+  bool initialized_;
```

Comment:
Why not just use `has_allocation_`

Reply:
We need a case for when CudaMemoryManager has been created but cnmemInit has not been called i.e. when memory pool sizes are 0.